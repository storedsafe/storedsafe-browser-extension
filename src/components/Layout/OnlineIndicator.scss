@import '../../variables';

.online-indicator {
  width: 0.8em;
  height: 0.8em;
  border-radius: 100%;
  transition: $transition-duration*3;
  transition-property: background-color, box-shadow;
  margin: 0.5em;

  &.online {
    background-color: $color-accent-0;
    box-shadow: 0 0 0.2em 0.2em $color-accent-1, 0 0 0.1em 0.1em rgba(0, 0, 0, 0.1) inset;
  }

  &.offline {
    background-color: $color-error-0;
    box-shadow: 0 0 0.2em 0.2em $color-error-1, 0 0 0.1em 0.1em rgba(0, 0, 0, 0.1) inset;
  }
}
