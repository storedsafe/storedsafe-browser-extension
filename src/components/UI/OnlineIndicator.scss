@import '../../variables';

.online-indicator {
  $size: 10px;

  width: $size;
  height: $size;
  border-radius: 100%;
  transition: $transition-duration*3;
  transition-property: background-color, box-shadow;
  margin: 0.5em;

  &.online {
    $shadow: $size / 4;
    background-color: $color-accent-0;
    box-shadow: 0 0 $shadow $shadow $color-accent-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }

  &.offline {
    $shadow: $size / 4;
    background-color: $color-danger-0;
    box-shadow: 0 0 $shadow $shadow $color-danger-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }
}
