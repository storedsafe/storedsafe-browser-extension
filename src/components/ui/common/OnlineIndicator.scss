@import '../../../variables';

.online-indicator {
  $size: 10px;
  padding: 0.5em;

  .online-indicator-icon {
    border-radius: 100%;
    width: $size;
    height: $size;
    transition: $transition-duration*3;
    transition-property: background-color, box-shadow;
  }

  &.online .online-indicator-icon {
    $shadow: $size / 4;
    background-color: $color-accent-0;
    box-shadow: 0 0 $shadow $shadow $color-accent-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }

  &.offline .online-indicator-icon {
    $shadow: $size / 4;
    background-color: $color-danger-0;
    box-shadow: 0 0 $shadow $shadow $color-danger-1, 0 0 $shadow / 2 $shadow / 2 rgba(0, 0, 0, 0.1) inset;
  }
}
