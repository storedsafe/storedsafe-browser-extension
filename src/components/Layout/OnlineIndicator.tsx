import React from 'react';
import './OnlineIndicator.scss';

interface OnlineIndicatorProps {
  online: boolean;
}

export const OnlineIndicator: React.SFC<OnlineIndicatorProps> = ({
  online,
}: OnlineIndicatorProps) => (
  <div className={`online-indicator ${online ? 'online' : 'offline'}`} />
);
