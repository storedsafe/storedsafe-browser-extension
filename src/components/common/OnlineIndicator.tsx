import React from 'react'
import './OnlineIndicator.scss'

interface OnlineIndicatorProps {
  online: boolean
}

export const OnlineIndicator: React.FunctionComponent<OnlineIndicatorProps> = ({
  online
}: OnlineIndicatorProps) => (
  <div className={`online-indicator ${online ? 'online' : 'offline'}`}>
    <div className="online-indicator-icon" />
  </div>
)
