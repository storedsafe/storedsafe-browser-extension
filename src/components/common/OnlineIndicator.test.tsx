import * as React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { OnlineIndicator } from './OnlineIndicator'
import pretty from 'pretty'

/**
 * Set up and tear down container to mount tested component in.
 * */
let container: Element = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

test('<OnlineIndicator />', () => {
  act(() => {
    render(<OnlineIndicator online={true} />, container)
  })
  expect(pretty(container.innerHTML)).toMatchSnapshot()

  act(() => {
    render(<OnlineIndicator online={false} />, container)
  })
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})
