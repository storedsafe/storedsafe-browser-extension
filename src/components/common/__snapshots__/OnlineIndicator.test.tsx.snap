// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnlineIndicator /> 1`] = `
"<div class=\\"online-indicator online\\">
  <div class=\\"online-indicator-icon\\"></div>
</div>"
`;

exports[`<OnlineIndicator /> 2`] = `
"<div class=\\"online-indicator offline\\">
  <div class=\\"online-indicator-icon\\"></div>
</div>"
`;
